src/compiler/crystal/codegen/debug.cr

```diff
@@ -43,35 +50,50 @@ module Crystal
       # from one module isn't incorrectly used in another module.
       debug_types_per_module =
         @debug_types_per_module ||=
-          {} of LLVM::Module => Hash(Type, LibLLVMExt::Metadata?)
+          {} of LLVM::Module => Hash(String, LibLLVMExt::Metadata?)
 
-      debug_types_per_module[@llvm_mod] ||= {} of Type => LibLLVMExt::Metadata?
+      debug_types_per_module[@llvm_mod] ||= {} of String => LibLLVMExt::Metadata?
     end
 
     def get_debug_type(type)
+      get_debug_type(type, type.to_s)
+    end
+
+    def get_debug_type(type, type_name : String? = type.to_s)
```

Comment:
```suggestion
    def get_debug_type(type, type_name : String)
```
ditto elsewhere

Reply:
`type_name : String = type.to_s` is mandatory as I explained to RX14 before.

Reply:
Hmm, why it's mandatory as u say? Regarding the discussion, I've only noticed https://github.com/crystal-lang/crystal/pull/8538#discussion_r385197891.

Reply:
In some cases new class name is not passed and needs to be passed via default value.
It needed for this case (at least):
```
    def create_debug_type(type : NilableType, type_name : String? = type.to_s)
      get_debug_type(type.not_nil_type, type_name)
    end
```
As you can see we pass type.not nil_type, but still assigns name of a NilableType

Reply:
Hmm, in the example you've given, the `type_name` argument is explicitly passed to `get_debug_type` call, so where's the problem then?

Reply:
You are right! I will cleanup code. Noone is calling `create_debug_type` directly anyway.